int32 axis_error
int32 axis_state